### PR TITLE
ENH: Use the CFIN dataset rather than the CENIR dataset.

### PR DESCRIPTION
**Pull request by jhlegarreta**
Following the discussion in #1291, use the ``CFIN`` dataset instead of
the ``CENIR`` dataset.

Also, make an explicit mention to the `MAPL` acronym so that the use of
the Laplacian-regularized MAP-MRI method seems no longer to be hidden.

### CODE REVIEW

**Comment by jhlegarreta** [comment]
@Garyfallidis @arokem 

I removed the 
```
For this example we select only the shell with b-values equal to the one of the Human Connectome Project (HCP).
```
sentence from the documentation since `read_cfin_dwi` does not accept the b-value list as an argument.

Finally, I got an error while debugging the example:
```
  File "C:/SDKs/dipy/doc/examples/reconst_mapmri.py", line 124, in <module>
    positivity_constraint=True)
  File "C:\SDKs\dipy\dipy\reconst\mapmri.py", line 251, in __init__
    'CVXPY package needed to enforce constraints')
ValueError: CVXPY package needed to enforce constraints
```
at
```
map_model_positivity_aniso = mapmri.MapmriModel(gtab,
                                                radial_order=radial_order,
                                                laplacian_regularization=False,
                                                positivity_constraint=True)
```

If CI builds do not report issues, I can assume this is a local issue, and that our `requirements.txt` does not need to be updated to include the `CVXPY` package. Correct me if I'm wrong.

**Comment by jhlegarreta** [comment]
Looks like CircleCI containers do have the `CVXPY` package installed. Any comments on requiring it?
 
I've noticed that the `CENIR` data has **also** data for **b-vals** **200** and **400** s/mm2. The **plots in the example will not show data for all data** since they only contain three subplots; **should we show all data, or just take into account only the 1000, 2000 and 3000** s/mm2, as the previous version was doing?

**Comment by skoudoro** [comment]
Can you review this PR @arokem?

**Comment by arokem** [comment]
Will do!

**Comment by arokem** [comment]
Moving to this new dataset will require some additional changes. See https://github.com/jhlegarreta/dipy/pull/1

**Comment by jhlegarreta** [comment]
Thanks for the review @arokem !

**Comment by arokem** [comment]
Thank you! I am +1 for merging this. If someone else could take a look, that would be great.

**Comment by RicciWoo** [comment]
After PR: [mapmri using cvxpy instead of cvxopt #1285](https://github.com/nipy/dipy/pull/1285), we are now using package CVXPY instead of CVXOPT, I think we need to update the comment on line 27-31 in reconst_mapmripy for this.

**Comment by skoudoro** [comment]
Hi @jhlegarreta, I got this error when I tried to run this example with the new dataset:
```
Traceback (most recent call last):
  File "reconst_mapmri.py", line 162, in <module>
    vmin=0, vmax=5e7)
  File "C:\Users\skoudoro\AppData\Local\Continuum\Anaconda3\envs\dipy-dev-3\lib\site-packages\matplotlib\__init__.py", line 1897, in inner
    return func(ax, *args, **kwargs)
  File "C:\Users\skoudoro\AppData\Local\Continuum\Anaconda3\envs\dipy-dev-3\lib\site-packages\matplotlib\axes\_axes.py", line 5124, in imshow
    im.set_data(X)
  File "C:\Users\skoudoro\AppData\Local\Continuum\Anaconda3\envs\dipy-dev-3\lib\site-packages\matplotlib\image.py", line 596, in set_data
    raise TypeError("Image data can not convert to float")
TypeError: Image data can not convert to float
```
I am working on Windows. Do you get this error ? Can you update the plotting part.
Thank you

**Comment by jhlegarreta** [comment]
@skoudoro thanks for the review. I did get the same error.

Please, check if commit 795aa4f solves the plot part issues.

Also, I am getting an error at
```
/dipy/doc/examples/reconst_mapmri.py", line 393, in <module>
    print('odf.shape (%d, %d, %d, %d)' % odf.shape)
```
telling me that
```
TypeError: not enough arguments for format string
```

Let's see what CI says, or whether you do get the same error; if you do, I'll amend with
```
print('odf.shape ({0})'.format(odf.shape))
```

BTW, although I know other parts of the code use `cvxpy`, in order to run the exampled I had to manually install it (pip was not listing it either as installed), so I'm wondering whether `cvxpy` should be added to the `requirements.txt` file.

May be this is a Windows-specific issue (?)

**Comment by jhlegarreta** [comment]
Looks like travis is happy; may be because no image is run on Windows machines (?). @skoudoro let me know when you check the patch set.

**Comment by arokem** [comment]
It's because the examples are not run as part of the CI. That would take
way too long.

**Comment by skoudoro** [comment]
Thanks @jhlegarreta. Today, I do not have access to my windows machine so I will try it tomorrow.

Concerning `cvxpy`, I think we want to let it as an optional requirement so do not add it to `requirement.txt`

**Comment by jhlegarreta** [comment]
@skoudoro 3eb7ddb was rebased on master.

As for the data region, just to make sure, you mean using a slice instead of a volume for the example now?

Regardless of the data region, not sure whether the result images are correct: I get all 3 images in `MAPMRI_maps_regularization.png` black.

**Comment by skoudoro** [comment]
> As for the data region, just to make sure, you mean using a slice instead of a volume for the example now?

`print(data.shape)` -> (96,96,19,496)
`data_small = data[40:65, 50:51, 35:60]` -> as you can see here `35:60`do not exist on the new dataset. In both case (old, and new dataset), it seems we work with one slice.

>  I get all 3 images in MAPMRI_maps_regularization.png black

In interactive mode, all results seem good when we display them, so maybe a saving issue, I will look at that

**Comment by jhlegarreta** [comment]
OK. Let me know if you are able to reproduce the issue (working on Win 10); if not, may be it's local to my environment and we can ignore it. Thanks Serge.

File: doc/examples/reconst_mapmri.py [hunk 1]
```diff
@@ -161,21 +163,23 @@
...
 
 ax2 = fig.add_subplot(1, 3, 2, title=r'RTOP - Positivity')
 ax2.set_axis_off()
-ind = ax2.imshow(mapfit_positivity_aniso.rtop()[:, 0, :].T,
-                 interpolation='nearest', origin='lower', cmap=plt.cm.gray,
-                 vmin=0, vmax=5e7)
+rtop_positivity = np.array(mapfit_positivity_aniso.rtop()[:, 0, :].T,
+                           dtype=float)
+ind = ax2.imshow(rtop_positivity, interpolation='nearest',
+                 origin='lower', cmap=plt.cm.gray, vmin=0, vmax=5e7)
```
Comment (skoudoro) [inline doc/examples/reconst_mapmri.py]: `vmin` and `vmax` are incorrect. This wrong range is the reason for your black image. Can you just remove this 2 parameters and let  `matplotlib` manage it

File: doc/examples/reconst_mapmri.py [hunk 2]
```diff
@@ -161,21 +163,23 @@
...
 
 ax3 = fig.add_subplot(1, 3, 3, title=r'RTOP - Both')
 ax3.set_axis_off()
-ind = ax3.imshow(mapfit_both_aniso.rtop()[:, 0, :].T,
-                 interpolation='nearest', origin='lower', cmap=plt.cm.gray,
-                 vmin=0, vmax=5e7)
+rtop_both = np.array(mapfit_both_aniso.rtop()[:, 0, :].T, dtype=float)
+ind = ax3.imshow(rtop_both, interpolation='nearest', origin='lower',
+                 cmap=plt.cm.gray, vmin=0, vmax=5e7)
```
Comment (skoudoro) [inline doc/examples/reconst_mapmri.py]: `vmin` and `vmax` are incorrect. This wrong range is the reason for your black image. Can you just remove this 2 parameters and let  `matplotlib` manage it

**Comment by skoudoro** [comment]
Due to the release, I will merge it and make the last correction on new PR.

Thank you @jhlegarreta for this PR and @arokem for this review!

**Pull request merged**